modelling/src/main/java/org/axonframework/modelling/command/AggregateAnnotationCommandHandler.java

```diff
@@ -334,6 +374,34 @@ public boolean canHandle(CommandMessage<?> message) {
         }
     }
 
+    private class AggregateCreateOrUpdateCommandHandler implements MessageHandler<CommandMessage<?>> {
+
+        private final MessageHandlingMember<? super T> handler;
+        private final Callable<T> factoryMethod;
+
+        public AggregateCreateOrUpdateCommandHandler(MessageHandlingMember<? super T> handler,
+                                                     Callable<T> factoryMethod) {
+            this.handler = handler;
+            this.factoryMethod = factoryMethod;
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Object handle(CommandMessage<?> command) throws Exception {
+            VersionedAggregateIdentifier iv = commandTargetResolver.resolveTarget(command);
+            Aggregate<T> inst = repository.loadOrCreate(iv.getIdentifier(), factoryMethod);
+            Object result = inst.handle(command);
+            assertThat(inst.identifier(), id -> id != null && id.equals(iv.getIdentifier()),
```

Comment:
@MGathier you mind if I ask the reasoning behind this assumption ?

Reply:
The check for the identifier is perfectly valid, the check it self is however not entirely correct. Luckily, this has already been resolved in [this](https://github.com/AxonFramework/AxonFramework/pull/1355/commits/d4b5ef9709ea0e2626d45bbfd0aace4db3c99825) commit.